src/client-rs/src/client/client.rs

```diff
@@ -40,21 +40,21 @@ pub trait Client: Sync + Send {
         R: Request + Send + Sync;
 }
 
-/// Client with socket connection
-pub struct RawClient {
+/// Socket with Versions
+pub struct VersionedSocket {
```

Comment:
Could you add some explanation in the doc as to what you mean by "versioned" socket? What is being specified by the version? Is it the contents being sent? Is there a schema associated with the socket?

Reply:
I will add more details.  Basically, sockets knows about list of supported API versions.